File: src/0/dp/1014/S1014_v1.go
func maxScoreSightseeingPair(values []int) int {
    res, imax := 0, 0
    for i := range values {
        res = max(res, imax + values[i])
        if values[i] > imax {
            imax = values[i]
        }
        imax--
    }
    return res
}

func max(x, y int) int {
    if x > y {
        return x
    }
    return y
}

File: src/0/dp/1014/S1014_v1.rs
impl Solution {
    pub fn max_score_sightseeing_pair(values: Vec<i32>) -> i32 {
        let mut imax = 0;
        let mut res = 0;
        for &i in values.iter() {
            res = res.max(imax + i);
            if i > imax {imax = i;}
            imax -= 1;
        }
        res
    }
}
